nhpy/run_full_results.py

```diff
@@ -255,33 +250,53 @@ def run_scenario_with_full_results(
         account_url = account_url or env_config["AZ_STORAGE_EP"]
         container_name = container_name or env_config["AZ_STORAGE_RESULTS"]
 
+    # Check that all required parameters are now set. If not, raise an error.
+    missing_params = [
+        param_name
+        for param_name, param_value in {
+            "api_url": api_url,
+            "api_key": api_key,
+            "account_url": account_url,
+            "container_name": container_name,
+        }.items()
+        if not param_value
+    ]
+    if missing_params:
+        raise EnvironmentVariableError(
+            missing_vars=missing_params,
+            message=f"Missing required parameters: {', '.join(missing_params)}",
+        )
+
     # Validate the results path format
     try:
         _extract_scenario_components(results_path=results_path)
     except ValueError as e:
         logger.error(f"run_scenario_with_full_results():Invalid results path: {e}")
         raise
 
-    if account_url and container_name:
-        # Load scenario parameters
-        params = _load_scenario_params(
-            results_path=results_path,
-            account_url=account_url,
-            container_name=container_name,
-        )
+    # Load scenario parameters
+    # At this point account_url and container_name are guaranteed to be non-None
+    # due to the checks above, but we need to help the type checker
+    params = _load_scenario_params(
+        results_path=results_path,
+        account_url=account_url if account_url is not None else "",
+        container_name=container_name if container_name is not None else "",
+    )
 
         # Prepare parameters for full results run
         mod_params = _prepare_full_results_params(params=params)
 
-        # Validate parameters against schema
-        _validate_params(mod_params)
-
-        container_metadata = _start_container(mod_params)
-
-        # Track container status
-        _track_container_status(container_metadata)
+    # Submit API request
+    # At this point api_url and api_key are guaranteed to be non-None
+    # due to the checks above, but we need to help the type checker
+    server_datetime = _submit_api_request(
+        params=mod_params,
+        api_url=api_url if api_url is not None else "",
+        api_key=api_key if api_key is not None else "",
+        timeout=Constants.TIMEOUT_SEC,
+    )
```

Comment:
?? Why are we moving back to using the API? We should be using nhp_aci and _start_container instead

Reply:
run_full_results shouldn't really be part of this PR at all